packages/sdk/src/plt/CborAccountAddress.ts

```diff
@@ -0,0 +1,297 @@
+ * [BCR-2020-007]{@link https://github.com/BlockchainCommons/Research/blob/master/papers/bcr-2020-007-hdkey.md}
+ *
+ * Example of CBOR diagnostic notation for an encoded account address:
+ * ```
+ * 40307({
+ *   1: 40305({1: 919}),
+ *   3: h'abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789'
+ * })
+ * ```
+ * Where 919 is the Concordium network identifier and the hex string is the raw account address.
+ *
+ * @param {Type} value - The token holder to convert to CBOR format.
+ * @throws {Error} - If an unsupported CBOR encoding is specified.
+ * @returns {Uint8Array} The CBOR encoded representation of the token holder.
+ */
+export function toCBOR(value: Type): Uint8Array {
+    return new Uint8Array(encode(toCBORValue(value)));
+}
+
+/**
+ * Registers a CBOR encoder for the CborAccountAddress type with the `cbor2` library.
+ * This allows CborAccountAddress instances to be automatically encoded when used with
+ * the `cbor2` library's encode function.
+ *
+ * @returns {void}
+ * @example
+ * // Register the encoder
+ * registerCBOREncoder();
+ * // Now CborAccountAddress instances can be encoded directly
+ * const encoded = encode(myCborAccountAddress);
+ */
+export function registerCBOREncoder(): void {
+    registerEncoder(CborAccountAddress, (value) => [TAGGED_ADDRESS, toCBORValue(value).contents]);
+}
+
+/**
+ * Decodes a CBOR-encoded token holder account into an {@linkcode Account} instance.
+ * @param {unknown} decoded - The CBOR decoded value, expected to be a tagged value with tag 40307.
+ * @throws {Error} - If the decoded value is not a valid CBOR encoded token holder account.
+ * @returns {Account} The decoded account address as a CborAccountAddress instance.
+ */
+function fromCBORValueAccount(decoded: unknown): CborAccountAddress {
+    // Verify we have a tagged value with tag 40307 (tagged-address)
+    if (!(decoded instanceof Tag) || decoded.tag !== TAGGED_ADDRESS) {
+        throw new Error(`Invalid CBOR encoded token holder account: expected tag ${TAGGED_ADDRESS}`);
+    }
+
+    const value = decoded.contents;
+
+    if (!(value instanceof Map)) {
+        throw new Error('Invalid CBOR encoded token holder account: expected a map');
+    }
+
+    // Verify the map corresponds to the BCR-2020-009 `address` format
+    const validKeys = [1, 2, 3]; // we allow 2 here, as it is in the spec for BCR-2020-009 `address`, but we don't use it
+    for (const key of value.keys()) {
+        validKeys.includes(key) || bail(`Invalid CBOR encoded token holder account: unexpected key ${key}`);
+    }
+
+    // Extract the token holder account bytes (key 3)
+    const addressBytes = value.get(3);
+    if (
+        !addressBytes ||
+        !(addressBytes instanceof Uint8Array) ||
+        addressBytes.byteLength !== AccountAddress.BYTES_LENGTH
+    ) {
+        throw new Error('Invalid CBOR encoded token holder account: missing or invalid address bytes');
```

Comment:
The whole file has several comments and error messages calling it `token holder`.

```suggestion
        throw new Error('Invalid CBOR encoded account address: missing or invalid address bytes');
```